common/sagas/gas.ts

```diff
@@ -0,0 +1,53 @@
+import { setGasEstimates, TypeKeys } from 'actions/gas';
+import { SagaIterator } from 'redux-saga';
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+import { AppState } from 'reducers';
+import { fetchGasEstimates, GasEstimates } from 'api/gas';
+import { gasPriceDefaults, gasEstimateCacheTime } from 'config';
+import { getEstimates } from 'selectors/gas';
+import { getOffline } from 'selectors/config';
+
+export function* setDefaultEstimates(): SagaIterator {
+  // Must yield time for testability
+  const time = yield call(Date.now);
+
+  yield put(
+    setGasEstimates({
+      safeLow: gasPriceDefaults.minGwei,
+      standard: gasPriceDefaults.default,
+      fast: gasPriceDefaults.default,
+      fastest: gasPriceDefaults.maxGwei,
+      isDefault: true,
+      time
+    })
+  );
+}
+
+export function* fetchEstimates(): SagaIterator {
+  // Don't even try offline
+  const isOffline: boolean = yield select(getOffline);
+  if (isOffline) {
+    yield call(setDefaultEstimates);
+    return;
+  }
+
+  // Cache estimates for a bit
+  const oldEstimates: AppState['gas']['estimates'] = yield select(getEstimates);
+  if (oldEstimates && oldEstimates.time + gasEstimateCacheTime > Date.now()) {
+    yield put(setGasEstimates(oldEstimates));
+    return;
+  }
+
+  // Try to fetch new estimates
+  try {
+    const estimates: GasEstimates = yield call(fetchGasEstimates);
```

Comment:
I would like to include validation on the `estimates` value to ensure that we aren't setting unexpectedly high default gas prices for users in the event of an error in the API response.

We could create a utility function `checkValidEstimate` that ensures the default value (`fast`) is not above e.g. 250 gwei, and do some additional loose checking on the min/max values.